### PR TITLE
Build StickyPopup components 

### PR DESCRIPTION
**Pull request by markcipolla**
REFS https://trello.com/c/zQvueItJ/1000-sticky-popup
REFS https://trello.com/c/yjOzW8cl/1008-build-sticky-popup-component-in-tcui

## Task: Adapt MUI's 'Snackbar' component to our colour themes and font styles to become "Sticky pop up"**

- MUI: https://material-ui.com/components/snackbars/ + https://material.io/components/snackbars/
- Our design: https://gallery.io/projects/MCHbtQVoQ2HCZWiQBR82z4LY/files/MCEJu8Y2hyDScXXP1OFK2xzMOO7rIvg1d1Y

## Background:
This update is one of many design recommendations for enhancing our email experience. The intention is that a snackbar would appear after 60-80% scroll on an article page. It could promote newsletter subscription today (purple), be reused for donations later in the year (red) and potentially be a consideration for republishing promos (default blue grey). Like all components, the snackbar should be device agnostic. 

## This PR

This PR creates a `StickyPopup` component. Under the hood, it builds a MUI `Snackbar` component, and automatically wraps the content in a `SnackbarContent` component; this allows us to wrap the whole thing in a `ThemeProvider` and get the styling we want.

It also adds a little space between all the `children` props; you pass the message sub-components (like `Typography` or `Button` components) in as children.

There are a few boolean props; `prominent` makes the spacing a bit larger, and `dismissable` includes a `StickyPopupDismiss` icon top-right of the `StickyPopup`.

## Example Usage:

```js
<ThemeProvider theme={defaultTheme()}>
  <StickyPopup
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'right',
    }}
    open={this.state.open}
    color="primary"
    onClose={this.handleClose}
    dismissable={true}
  >
    <Typography variant="subtitle1">
      Join over 100,000 Australians who value free evidence-based news.
    </Typography>
    <Button variant="contained" onClick={action("click")}>
      Sign up today
    </Button>
  </StickyPopup>
</ThemeProvider />
```

## Differences between TC & MUI

- Unlike MUI's snackbar, this should support dismissable behaviour when required

## Screenshots
popups

## To Do in upcoming PRs
- [ ] Update TCUI font sizes for Typography components. This will allow us to more closely match designs by @ZoeJazz 
- [ ] Optionally disable ClickAwayHandlers (you can click anywhere outside the `StickyPopup` and it'll dismiss it. Maybe we want make that optional?
- [ ] Add functionality to control slide direction, which we will definitely use to slide in from the bottom on mobiles &mdash; https://material-ui.com/components/snackbars/#control-slide-direction
- [ ] Add functionality to control transition type &mdash; https://material-ui.com/components/snackbars/#change-transition

### CODE REVIEW

File: src/StickyPopup/StickyPopup.jsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Snackbar from '@material-ui/core/Snackbar'
+import SnackbarContent from '@material-ui/core/SnackbarContent'
+import withStyles from '@material-ui/core/styles/withStyles'
+
+import StickyPopupDismiss from './StickyPopupDismiss'
+
+// Like https://github.com/brunobertolini/styled-by
+const styledBy = (property, mapping) => props => mapping[props[property]]
```
Comment (nullobject) [inline src/StickyPopup/StickyPopup.jsx]: This is handy 👍

Comment (markcipolla) [reply src/StickyPopup/StickyPopup.jsx]: Now I think of it, I've used it in a few components. Should extract it somewhere like `utils`.

**Comment by ZoeJazz** [comment]
Great work @markcipolla :)

File: src/StickyPopup/StickyPopup.jsx (hunk 1 above)
Comment (markcipolla) [reply src/StickyPopup/StickyPopup.jsx]: Done.

**Comment by markcipolla** [comment]
> This is great 👍
> 
> My only comment is that I find the example buttons a bit verbose. How about just: "Show Donate StickyPopup", etc?

Done.

**Comment by jameshill** [comment]
🎉

**Comment by jameshill** [comment]
Fantastic work!

**Pull request merged**